src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -288,6 +291,29 @@ def get_composite_package_identifier(self, package, package_version):
 
     def install_updates_fail_safe(self, excluded_packages):
         return
+
+    def accept_eula_for_patches(self):
+        """ Accepts eula for patches based on the config provided by customers """
+        if not os.path.exists(Constants.Paths.EULA_SETTINGS):
+            self.composite_logger.log_warning("NOT accepting EULA for any patch as no corresponding EULA Settings found on the VM")
+            return
+
+        try:
+            eula_settings = json.loads(self.env_layer.file_system.read_with_retry(Constants.Paths.EULA_SETTINGS, raise_if_not_found=False) or 'null')
+
+            if eula_settings is not None \
+                    and Constants.EulaSettings.ACCEPT_EULA_FOR_ALL_PATCHES in eula_settings \
+                    and eula_settings[Constants.EulaSettings.ACCEPT_EULA_FOR_ALL_PATCHES] is True:
+                self.composite_logger.log_debug("Accepting EULA for all patches as per the EULA setting found on VM")
+                self.single_package_upgrade_simulation_cmd += " ACCEPT_EULA=Y"
+                self.single_package_dependency_resolution_template += " ACCEPT_EULA=Y"
+                self.single_package_upgrade_cmd += " ACCEPT_EULA=Y"
+            else:
+                self.composite_logger.log_warning("NOT accepting EULA for any patch as no corresponding acceptance for EULA found on the VM")
+                return
+        except Exception as error:
```

Comment:
add test, possible to consider using self.assertRaises(Exception....

Reply:
Done